go-controller/pkg/util/net_linux.go

```diff
@@ -335,14 +335,27 @@ func LinkRoutesAddOrUpdateSourceOrMTU(link netlink.Link, gwIP net.IP, subnets []
 	return nil
 }
 
-// LinkRouteGet gets a route for the given subnet with the specified gwIPStr
+// LinkRouteGetWithSubnetIP gets a route for the given subnet dst IP
 // returns nil if route is not found
-func LinkRouteGet(link netlink.Link, gwIP net.IP, subnet *net.IPNet) (*netlink.Route, error) {
-	routeFilter := &netlink.Route{Dst: subnet, LinkIndex: link.Attrs().Index}
-	filterMask := netlink.RT_FILTER_DST | netlink.RT_FILTER_OIF
-	routes, err := netLinkOps.RouteListFiltered(getFamily(gwIP), routeFilter, filterMask)
+func LinkRouteGetWithSubnetIP(link netlink.Link, gwIP net.IP, subnet *net.IPNet) (*netlink.Route, error) {
+	routes, err := filterRoutesBasedOnDst(link, gwIP, subnet)
+	if err != nil {
+		return nil, err
+	}
+	for _, route := range routes {
+		if route.Dst.IP.Equal(subnet.IP) {
```

Comment:
nit: shall it also compare the dst mask for accuracy?

Reply:
Good point, nothing bad in that afaiu.

One thing I wonder though... I think matching the route dst might be redundant, since, afaiu, we're already filtering these routes according to the [route filter](https://github.com/ovn-org/ovn-kubernetes/pull/3136/files#diff-462c1c75c92761bec5429ecd1a006c4228ecfb2f1a5694fd98a69ac9efc6dd27R369) - which filters based on the dst subnet.

@trozet / @zshi-redhat

Reply:
@maiqueb I think you're right. Given there is only one route matching service subnet, maybe we can just remove the gwIP match and return `routers[0]`?

Reply:
@zshi-redhat / @trozet I've taken the liberty of simplifying the route filtering stuff a bit while waiting for the review. 

Let me know what you think.

Reply:
Hm, I think I need to dig a bit deeper. While looking into our [code](https://github.com/containernetworking/plugins/blob/main/pkg/ip/utils_linux.go#L67) and also the[ netlink's unit tests](https://github.com/vishvananda/netlink/blob/main/route_test.go#L892), I get the impression the `RT_FILTER_GW` is not required.

While I am nearly sure matching on the GW as we currently do is redundant, I need to better understand this.